### PR TITLE
Issue1435 Anonymize usernames in answer distribution reports.

### PR DESCRIPTION
**Pull request by jpaille**
Using the private key defined in settings.
Ticket https://fun.plan.io/issues/1435.

### CODE REVIEW

File: fun/envs/common.py [hunk 1]
```diff
@@ -220,3 +220,5 @@ def default_cache_configuration(key_prefix):
         "LOCATION": ORA2_FILEUPLOAD_CACHE_ROOT
     }
 }
+
+ANONYMIZATION_KEY = """179UeqhbzllzpGxqj*lu"""
```
Comment (regisb) [inline fun/envs/common.py]: ANONYMIZATION_KEY = """dummykey"""

File: course_dashboard/reports_manager/tests/test_tasks.py [hunk 2]
```diff
@@ -60,8 +62,9 @@ def _read_report_file(self, filepath):
     def _create_response_row(self):
         user_profile = UserProfile.objects.get(user__username=self.username)
         response_row = [unicode(x) for x in
-                        [user_profile.user.id, user_profile.gender, user_profile.year_of_birth,
-                         user_profile.level_of_education]] + [OPTION_1, OPTION_2]
+                        [hmac.new(settings.ANONYMIZATION_KEY, user_profile.user.username, hashlib.sha256).hexdigest(),
```
Comment (regisb) [inline course_dashboard/reports_manager/tests/test_tasks.py]: The encryption step is duplicated. You should move it to an imported module.

File: course_dashboard/reports_manager/tests/test_tasks.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
-import os
+import csv
+import hashlib
+import hmac
 import json
+import os
```
Comment (regisb) [inline course_dashboard/reports_manager/tests/test_tasks.py]: Why import os?

File: course_dashboard/reports_manager/tasks.py [hunk 4]
```diff
@@ -48,7 +49,7 @@ def fetch_user_profile_data(student_module):
     """
     user = student_module.student
     user_profile = UserProfile.objects.get(user=user)
-    data_row = [unicode(user.id),
+    data_row = [hmac.new(settings.ANONYMIZATION_KEY, user.username, hashlib.sha256).hexdigest(),
```
Comment (regisb) [inline course_dashboard/reports_manager/tasks.py]: Why not use the newly-created function?

Comment (jpaille) [inline course_dashboard/reports_manager/tasks.py]: my bad

**Comment by regisb** [comment]
lgtm

**Pull request closed without merging**